### ISSUE TITLE
strategies for long, thin, diagonal objects?

### ISSUE THREAD

**Issue by waspinator** [matterport/Mask_RCNN#853]
I've had success training on objects which fill in a large portions of their bounding boxes. For long, thin, diagonal objects that only fill in a small portion of their bonding boxes, segmentation doesn't work as well.

https://github.com/matterport/Mask_RCNN/issues/768 appears to have a similar issue. 

![stems](https://i.imgur.com/Rxmxxvq.jpg)

**Comment by captainst** [comment]
Hi there,
Got a similar problem here:
https://stackoverflow.com/questions/55389081/mask-rcnn-can-not-correctly-mask-long-thin-objects-like-animal-tail
Still can't figure out how to tune the hyper parameters. I plan to increase the training data in some way.

**Comment by mashuai191** [comment]
i'm struggle for the same issue. Any finding or solutions for fixing it?

**Comment by mashuai191** [comment]
the default mask shape is 28x28, and i think the shrink cause some loss, so after changing it to 56x56, it fixed the my problem. please see the following parameter i set in config. Note that the   MASK_SHAPE should be 2x MASK_POOL_SIZE  in number.

    POOL_SIZE = 14
    MASK_POOL_SIZE = 28
    MASK_SHAPE = [56, 56]

**Comment by jasdal365** [comment]
@mashuai191 can you please share your code? I coulnd't change POOL_SIZE because of some dimensional error. If I increase the MASK_SHAPE, all losses containing "mrcnn" are constantely zero. So I think the computation doesn't work anymore

**Comment by 1chimaruGin** [comment]
@waspinator Can I see your current config?

**Comment by kimile599** [comment]
> I've had success training on objects which fill in large portions of their bounding boxes. For long, thin, diagonal objects that only fill in a small portion of their bonding boxes, segmentation doesn't work as well.
> 
> Are there any strategies I could use to help segment these types of objects? My entire dataset is already diagonal objects, so I don't think using augmentation to rotate the objects would help in this case.
> 
> #768 appears to have a similar issue.
> 
> ![stems](https://camo.githubusercontent.com/85b03dab2e6b179268c92aa759cfbb91113d6597/68747470733a2f2f692e696d6775722e636f6d2f52786d787876712e6a7067)

In current cases, does your val_loss converging. In my case, i can get ok total loss, but my val loss does't converge at all.

**Comment by kimile599** [comment]
> the default mask shape is 28x28, and i think the shrink cause some loss, so after changing it to 56x56, it fixed the my problem. please see the following parameter i set in config. Note that the MASK_SHAPE should be 2x MASK_POOL_SIZE in number.
> 
> ```
> POOL_SIZE = 14
> MASK_POOL_SIZE = 28
> MASK_SHAPE = [56, 56]
> ```

Hey man, how is your val_loss looks and what's your sample number or dataset number?

**Comment by mashuai191** [comment]
i haven't touch my code for long time as i finally changed to detectron2, but maybe you can find something useful in my previous repo https://github.com/mashuai191/Mask_RCNN 
@kimile599 @jasdal365

**Comment by kimile599** [comment]
> i haven't touch my code for long time as i finally changed to detectron2, but maybe you can find something useful in my previous repo https://github.com/mashuai191/Mask_RCNN
> @kimile599 @jasdal365

thank you for the reply, i will check. BTW, do you mind adding you on wechat

**Comment by mashuai191** [comment]
> > i haven't touch my code for long time as i finally changed to detectron2, but maybe you can find something useful in my previous repo https://github.com/mashuai191/Mask_RCNN
> > @kimile599 @jasdal365
> 
> thank you for the reply, i will check. BTW, do you mind adding you on wechat

my wechat is 13880764134.
